pkg/filter/proxy/proxy.go

```diff
@@ -348,6 +379,17 @@ func (b *Proxy) handle(ctx context.HTTPContext) (result string) {
 		return ""
 	}
 
+	super := b.filterSpec.Super()
+	if super != nil {
+		tracing := super.Tracing()
+		if !tracing.IsNoopTracer() && !b.client.IsTracingInitialized() {
+			b.client.UpdateTracing(tracing)
+		} else if tracing.IsNoopTracer() && b.client.IsTracingInitialized() {
+			// if HTTPServer is updated and there is no longer tracing configured
+			b.reload()
+		}
```

Comment:
please check if there are race conditions.

Reply:
Now the proxy.client is protected by mutex